cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/service/CloudbreakResourceNameService.java

```diff
@@ -119,6 +120,8 @@ protected String appendHash(String name, Date timestamp) {
         DateFormat df = new SimpleDateFormat(DATE_FORMAT);
         return Joiner.on("").join(name, DELIMITER, df.format(timestamp));
     }
-}
-
 
+    protected String appendHash(String name, String toBeHashed) {
+        return Joiner.on("").join(name, DELIMITER, DigestUtils.md5DigestAsHex(toBeHashed.getBytes()).substring(0, DATE_FORMAT.length()));
```

Comment:
I would pass the cut length as param instead of always taking DATE_FORMAT.length